src/libre/print/ebnf.c

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright 2018 Katherine Flavel
+ *
+ * See LICENCE for the full copyright terms.
+ */
+
+#include <stdio.h>
+#include <ctype.h>
+
+#include <print/esc.h>
+
+#include "../re_ast.h"
+#include "../re_char_class.h"
+#include "../print.h"
+
+static void
+re_flags_print(FILE *f, enum re_flags fl);
+
+static void
+cc_pp_iter(FILE *f, const struct fsm_options *opt,
+	struct re_char_class_ast *n);
+
+static void
+pp_iter(FILE *f, const struct fsm_options *opt, struct ast_expr *n);
+
+static int
+atomic(struct ast_expr *n)
+{
+	switch (n->t) {
+	case AST_EXPR_EMPTY:
+	case AST_EXPR_LITERAL:
+	case AST_EXPR_ANY:
+	case AST_EXPR_CHAR_CLASS:
+	case AST_EXPR_GROUP:
+		return 1;
+
+	case AST_EXPR_REPEATED:
+	case AST_EXPR_CONCAT:
+	case AST_EXPR_ALT:
+	case AST_EXPR_MANY:
+		return 0;
+
+	case AST_EXPR_FLAGS:
+		return 0; /* XXX */
```

Comment:
Does the `XXX` here indicate you still mean to change this? `0` seems like a reasonable return value.
I think of the flag nodes as being purely metadata. I might be able to do away with them entirely, once the anchoring is fully working, because there's also node flags used for tracking the anchoring-related state.

Reply:
It means I question the line - in this case I think `0` probably makes sense, but something feels wrong. I don't know what.